src/org/mozilla/javascript/ScriptableObject.java

```diff
@@ -1581,7 +1581,7 @@ public void defineOwnProperty(Context cx, Object id, ScriptableObject desc) {
      * @param desc the new property descriptor, as described in 8.6.1
      * @param checkValid whether to perform validity checks
      */
-    protected void defineOwnProperty(
+    protected boolean defineOwnProperty(
```

Comment:
Why the change from void to boolean returnType? AFAICT all implementations/overrides of this method only ever return true

Reply:
Mmm, found that the returnType is used in (at least) the reflectImpl.

I wonder if this is the way to go about it: the spec says defineProperty doesn't return a value, it just throws if there's an issue with defining the property

So I wonder if instead of changing the returnType of the method and always returning true, it would not be better to leave the existing impl. as is and in the Reflect.defineProperty just add a `return true` after the call to defineProperty, as you already catch the exception and do a return false there

Reply:
https://tc39.es/ecma262/#table-essential-internal-methods